src/color.lisp

```diff
@@ -14,11 +14,38 @@
   ((red :initform 0.0 :accessor color-red :initarg :red)
    (green :initform 0.0 :accessor color-green :initarg :green)
    (blue :initform 0.0 :accessor color-blue :initarg :blue)
-   (hue :initform 0.0 :accessor color-hue :initarg :hue)
-   (saturation :initform 0.0 :accessor color-saturation :initarg :saturation)
-   (brightness :initform 0.0 :accessor color-brightness :initarg :brightness)
+   (hue :initform 0.0 :reader color-hue :initarg :hue)
```

Comment:
@TatriX Great catch. Sorry for not answering earlier.

I'm now realizing that rgb updates are not updating hsb either.
What do you think about making the code a bit less repetitive by keeping accessors and using the `:AFTER` qualifier? Something like:

```lisp
(defun update-rgb (color)
  (with-slots (red green blue hue saturation brightness) color
    (destructuring-bind (r g b) (hsb-to-rgb hue saturation brightness)
      (setf red r
            green g
            blue b))))

(defun update-hsb (color)
  (with-slots (red green blue hue saturation brightness) color
    (destructuring-bind (h s b) (rgb-to-hsb red green blue)
      (setf hue h
            saturation s
            brightness b))))

(defmethod (setf color-red) :after (value color)
  (update-hsb color))

(defmethod (setf color-green) :after (value color)
  (update-hsb color))

(defmethod (setf color-blue) :after (value color)
  (update-hsb color))

(defmethod (setf color-hue) :after (value color)
  (update-rgb color))

(defmethod (setf color-saturation) :after (value color)
  (update-rgb color))

(defmethod (setf color-brightness) :after (value color)
  (update-rgb color))
``` 

or maybe for even less verbosity

```lisp
(defmacro add-updater-to-accessor (accessor updater)
  `(defmethod (setf ,accessor) :after (value color)
     (,updater color)))

(add-updater-to-accessor color-red update-hsb)
(add-updater-to-accessor color-green update-hsb)
(add-updater-to-accessor color-blue update-hsb)
(add-updater-to-accessor color-hue update-rgb)
(add-updater-to-accessor color-saturation update-rgb)
(add-updater-to-accessor color-brightness update-rgb)
```

?

Reply:
That looks even better!